ams/optimizer/src/main/java/com/netease/arctic/optimizer/KubernetesOptimizerContainer.java

```diff
@@ -0,0 +1,87 @@
+package com.netease.arctic.optimizer;
+
+import com.netease.arctic.ams.api.resource.Resource;
+import com.netease.arctic.optimizer.util.K8sUtil;
+import com.netease.arctic.optimizer.util.PropertyUtil;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Map;
+
+public class KubernetesOptimizerContainer extends AbstractResourceContainer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KubernetesOptimizerContainer.class);
+
+  public static final String MEMORY_PROPERTY = "memory";
+  public static final String CPU_PROPERTY = "cpu";
+  public static final String NAMESPACE = "namespace";
+  public static final String IMAGE = "image";
+  public static final String KUBE_CONFIG_PATH = "kube-config-path";
+
+  public static final String NAME = "amoro-optimizer";
+  @Override
+  public void requestResource(Resource resource) {
+    // generate pod start args
+    long memoryPerThread = Long.parseLong(PropertyUtil.checkAndGetProperty(resource.getProperties(),
+        MEMORY_PROPERTY));
+    long memory = memoryPerThread * resource.getThreadCount();
+    // point at amoro home in docker image
+    String startUpArgs = String.format("/usr/local/amoro/bin/localOptimize.sh %s %s", memory,
+        super.buildOptimizerStartupArgsString(resource));
+    String[] cmd = { startUpArgs};
+    LOG.info("Starting k8s optimizer using k8s client with start command : {}", startUpArgs);
+    // start k8s job using k8s client
+    String kubeConfigPath = PropertyUtil.checkAndGetProperty(getContainerProperties(), KUBE_CONFIG_PATH);
+    String kubeConfig = getKubeConfigContent(kubeConfigPath);
+    String namespace = PropertyUtil.checkAndGetProperty(resource.getProperties(),
+        NAMESPACE);
```

Comment:
Can we set a default value for Namespace? If AMS is deployed in K8S, we can use the same Namespace as AMS. If AMS is outside the cluster, we can use 'Default' as the default Namespace.